### PR TITLE
Resize Bilinear Tensor Host

### PR DESCRIPTION
**Pull request by fiona-gladwin**
- Add tensor API for resize bilinear interpolation with Host backend
- Test suite changes to accept interpolation param and added test cases for resize.

### CODE REVIEW

File: src/modules/rppt_tensor_geometric_augmentations.cpp [hunk 1]
```diff
@@ -346,3 +346,66 @@ RppStatus rppt_warp_affine_gpu(RppPtr_t srcPtr,
...
+                           RppPtr_t dstPtr,
+                           RpptDescPtr dstDescPtr,
+                           RpptImagePatchPtr dstImgSizes,
+                           RpptInterpolationType interpolationType,
+                           RpptROIPtr roiTensorPtrSrc,
+                           RpptRoiType roiType,
+                           rppHandle_t rppHandle)
+{
+    RppLayoutParams srcLayoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+
```
Comment (r-abishek) [inline src/modules/rppt_tensor_geometric_augmentations.cpp:363]: remove extra line

File: src/modules/cpu/kernel/resize.hpp [hunk 2]
```diff
@@ -0,0 +1,1109 @@
...
+    roiPtrDefault->xywhROI.xy.x = 0;
+    roiPtrDefault->xywhROI.xy.y = 0;
+    roiPtrDefault->xywhROI.roiWidth = srcDescPtr->w;
+    roiPtrDefault->xywhROI.roiHeight = srcDescPtr->h;
+
+omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
```
Comment (r-abishek) [inline src/modules/cpu/kernel/resize.hpp:26]: L26-L52 could be replaced now with someething like 21-L23 here - https://github.com/r-abishek/rpp/blob/ar/opt_resize_bilinear_host/src/modules/cpu/kernel/brightness.hpp#L21-L23

File: src/modules/cpu/kernel/resize.hpp [hunk 3]
```diff
@@ -0,0 +1,1109 @@
...
+RppStatus resize_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                   RpptDescPtr srcDescPtr,
+                                   Rpp8u *dstPtr,
+                                   RpptDescPtr dstDescPtr,
+                                   RpptImagePatchPtr dstImgSize,
+                                   RpptROIPtr roiTensorPtrSrc,
+                                   RpptRoiType roiType,
+                                   RppLayoutParams srcLayoutParams)
+{
+    RpptROI roiDefault;
```
Comment (r-abishek) [inline src/modules/cpu/kernel/resize.hpp:14]: L14-L20 could be replaced too like - https://github.com/r-abishek/rpp/blob/ar/opt_resize_bilinear_host/src/modules/cpu/kernel/brightness.hpp#L15

File: src/modules/cpu/kernel/resize.hpp [hunk 4]
```diff
@@ -0,0 +1,1109 @@
...
+                srcRowPtrsForInterp[1]  = srcRowPtrsForInterp[0] + srcDescPtr->strides.hStride;           // srcPtrBottomRow for bilinear interpolation
+                pWeightParams[0] = _mm256_set1_ps(weightParams[0]);
+                pWeightParams[1]  = _mm256_set1_ps(weightParams[1]);
+                pDstLoc = pDstLocInit;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m256 pSrc[12], pDst[3];
+                    compute_resize_src_loc_avx(pDstLoc, pWRatio, pWidthLimit, srcLocationColumnArray, &pWeightParams[2], pWOffset, true);
```
Comment (r-abishek) [inline src/modules/cpu/kernel/resize.hpp:108]: For sections like these from L108-L112, you may want to leave a brief comment on the right for better readability - saying what each helper does

File: src/modules/cpu/kernel/resize.hpp [hunk 5]
```diff
@@ -0,0 +1,1109 @@
...
+            dstPtrRow = dstPtrChannel;
+
+            for(int dstLocRow = 0; dstLocRow < dstImgSize[batchCount].height; dstLocRow++)
+            {
+                Rpp8u *dstPtrTemp;
+                dstPtrTemp = dstPtrRow;
+
+                Rpp8u *srcRowPtrsForInterp[6];
+                compute_resize_src_loc(dstLocRow, hRatio, heightLimit, srcLocationRow, &weightParams[0], hOffset);
+                srcRowPtrsForInterp[0] = srcPtrChannel + srcLocationRow * srcDescPtr->strides.hStride;    // srcPtrTopRowR for bilinear interpolation
```
Comment (r-abishek) [inline src/modules/cpu/kernel/resize.hpp:145]: Please modularize if L145-L150 is repeated multiple times in this file. You could use a single templated helper for this one.

File: src/modules/cpu/kernel/resize.hpp [hunk 6]
```diff
@@ -0,0 +1,1109 @@
...
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        srcPtrChannel = srcPtrImage + (roiPtr->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr->xywhROI.xy.x * srcLayoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength = dstImgSize[batchCount].width & ~7;
+        __m256 pWRatio = _mm256_set1_ps(wRatio);
+        __m256 pWOffset = _mm256_set1_ps(wOffset);
+        __m256 pWidthLimit = _mm256_set1_ps((float)widthLimit);
+        __m256 pDstLocInit = _mm256_setr_ps(0, 1, 2, 3, 4, 5, 6, 7);
```
Comment (r-abishek) [inline src/modules/cpu/kernel/resize.hpp:75]: Please make this a global constant if you are not modifying its value elsewhere.

**Review by r-abishek — CHANGES_REQUESTED** [review]
Please see the review comments and make the necessary changes.

File: src/modules/cpu/kernel/resize.hpp (hunk 4 above)
Comment (fiona-gladwin) [reply src/modules/cpu/kernel/resize.hpp]: I have added the comments please check

File: src/modules/cpu/kernel/resize.hpp (hunk 5 above)
Comment (fiona-gladwin) [reply src/modules/cpu/kernel/resize.hpp]: Done

**Comment by fiona-gladwin** [comment]
@r-abishek I have made the changes with respect to your comments please check.

**Review by r-abishek — APPROVED** [review]
Looks good

**Pull request merged**